cli/server/getDatasetHelpers.js

```diff
@@ -47,18 +47,25 @@ const interpretRequest = (req) => {
 
 /**
  * Given a request, does the dataset exist?
- * In the future, if there is no exact match but a partial one we
- * should extend this. E.g. `["flu"]` -> `["flu", "h3n2", "ha", "3y"]`
- * In that case, we should utilise `res.redirect` as we do in the nextstrain.org server
- * @throws
  */
-const extendDataPathsToMatchAvailable = (info, availableDatasets) => {
-  const requestStrToMatch = info.parts.join("/"); // TO DO
-  /* TODO currently there must be an _exact_ match in the available datasets */
-  if (!availableDatasets.map((d) => d.request).includes(requestStrToMatch)) {
-    throw new Error(`${requestStrToMatch} not in available datasets`);
+const extendDataPathsToMatchAvailable = (res, info, availableDatasets) => {
```

Comment:
This is great! Could you please rename the function to better reflect the functionality -- along the lines of https://github.com/nextstrain/auspice/commit/fc824fcc6736a8263c3d8f7d9d47a78d0b3db352 (can just cherry-pick that if you're happy with it)

Reply:
The cherry-pick merged without any conflict so I kept it 👍

Reply:
For the rest, the new commit should take care of the changes requested. 

I wrote in a slightly different, more compact form but if you prefer I can rewrite it with a for cycle, it may be better for readability.